File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/repository/ITestRepository.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface ITestRepository <ID, T> : IRepository<ID, T> where T : HasId<ID>
    {
        IEnumerable<T> findAllTestsForParticipant(int id);
    }
}

File: Java_ChildrenCompetition/src/main/java/repository/ITestRepository.java
package repository;

import model.Test;

//public interface ITestRepository<ID, T extends HasId<ID>> extends IRepository<ID, T>{
//    Iterable<T> findAllTestsForParticipant(int id);
//}

public interface ITestRepository extends IRepository<Integer, Test>{
    Iterable<Test> findAllTestsForParticipant(int id);
}